src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { client } from '../../utils/fetchClient';
+import { URL, USER_ID } from '../../utils/Url';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setTodos: (a: Todo[] | any) => void;
+  setErrorMessage: (a: string) => void;
+  todos: Todo[];
+};
+
+export const TodoApp: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setErrorMessage,
+}) => {
+  const [value, setValue] = useState<string>('');
+
+  const handleAddTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newValue: string = event.target.value;
+
+    setValue(newValue);
+  };
+
+  const addNewTodo = async () => {
+    if (value.trim() === '') {
+      return;
+    }
+
+    try {
+      const newTodoData = {
+        title: value,
+        completed: false,
+        userId: USER_ID,
+      };
+
+      const addedTodo = await client.post<Todo>(URL, newTodoData);
+
+      setTodos((prevTodos: Todo[]) => [...prevTodos, addedTodo]);
+
+      setValue('');
+    } catch (error) {
+      setErrorMessage('Unable to add todo');
+      throw new Error('Unable to add todo');
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    addNewTodo();
+  };
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      handleSubmit(event as unknown as React.FormEvent<HTMLFormElement>);
+    }
+  };
+
+  const activeTodos = todos.map((todo: Todo) => !todo.completed);
```

Comment:
Your 'activeTodos' assignment is incorrect. The map function returns an array of boolean values, not an array of active todos. Use 'filter' instead of 'map' to get an array of active todos.